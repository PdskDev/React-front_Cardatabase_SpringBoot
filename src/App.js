import './App.css';
import { AppBar, Toolbar } from '@mui/material';
import Typography from '@mui/material/Typography';

import Login from './components/Login';

function App() {
  return (
    <div className="App">
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6">Carshop</Typography>
        </Toolbar>
      </AppBar>
      <Login />
    </div>
  );
}

export default App;
