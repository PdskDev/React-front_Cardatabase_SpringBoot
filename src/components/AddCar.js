import React, { useState } from 'react';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogTitle from '@mui/material/DialogTitle';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Stack from '@mui/material/Stack';
import Box from '@mui/material/Box';
import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';

function AddCar(props) {
  const [open, setOpen] = useState(false);
  const [car, setCar] = useState({
    brand: '',
    model: '',
    color: '',
    buildYear: '',
    price: '',
    registerNumber: '',
  });

  //Open the modal form
  const handleClickOpen = () => {
    setOpen(true);
  };

  //Close the modal form
  const handleClose = () => {
    setOpen(false);
  };

  //handle change input values
  const handleChange = (event) => {
    setCar({ ...car, [event.target.name]: event.target.value });
  };

  //Save new car
  const handleSave = () => {
    props.addCar(car);
    handleClose();
  };

  return (
    <div>
      <Button variant="contained" color="primary" onClick={handleClickOpen}>
        <AddCircleOutlineIcon />
        New Car
      </Button>
      <Dialog open={open} onClose={handleClose}>
        <Box sx={{ width: 500 }}>
          <DialogTitle>New Car</DialogTitle>
          <DialogContent>
            <Stack spacing={2} mt={1}>
              <TextField
                label="Brand"
                name="brand"
                autoFocus
                variant="standard"
                value={car.brand}
                onChange={handleChange}
              />
              <TextField
                label="Model"
                name="model"
                autoFocus
                variant="standard"
                value={car.model}
                onChange={handleChange}
              />
              <TextField
                label="Color"
                name="color"
                autoFocus
                variant="standard"
                value={car.color}
                onChange={handleChange}
              />
              <TextField
                label="Year"
                name="buildYear"
                autoFocus
                variant="standard"
                value={car.buildYear}
                onChange={handleChange}
              />
              <TextField
                label="Register Number"
                name="registerNumber"
                autoFocus
                variant="standard"
                value={car.registerNumber}
                onChange={handleChange}
              />
              <TextField
                label="Price"
                name="price"
                autoFocus
                variant="standard"
                value={car.price}
                onChange={handleChange}
              />
            </Stack>
          </DialogContent>
          <DialogActions>
            <Button variant="contained" color="error" onClick={handleClose}>
              Cancel
            </Button>
            <Button variant="contained" color="primary" onClick={handleSave}>
              Save
            </Button>
          </DialogActions>
        </Box>
      </Dialog>
    </div>
  );
}

export default AddCar;
