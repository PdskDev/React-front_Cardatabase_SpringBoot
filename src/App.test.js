import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import TestRender from 'react-test-renderer';
import AddCar from './components/AddCar';

test('open add car modal form', async () => {
  render(<App />);
  fireEvent.click(screen.getByText('New Car'));
  expect(screen.getByRole('dialog')).toHaveTextContent('New Car');
});

test('renders a snapshot', () => {
  const tree = TestRender.create(<AddCar />).toJSON();
  expect(tree).toMatchSnapshot();
});
