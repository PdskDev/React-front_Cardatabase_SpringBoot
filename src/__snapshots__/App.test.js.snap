// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddCircleOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    New Car
  </button>
</div>
`;
